### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by petewanca**
# Description

Adds:
- user auth routes, controller and service level
- input validation, server and (minimal) client side
- changes some book functions to arrow functions

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -1,38 +1,53 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
+import React, { useState, useEffect, useContext } from 'react';
 import './App.css';
+import { Nav } from './components//Nav/Nav';
```
Comment (acd37) [inline client/src/App.js:3]: malformed link `//Nav`

File: src/models/User.js [hunk 2]
```diff
@@ -1,5 +1,29 @@
 module.exports = function (sequelize, Sequelize) {
-  const User = sequelize.define('user', {});
+  const User = sequelize.define('user', {
+    id: {
+      type: Sequelize.UUID,
+      primaryKey: true,
+      defaultValue: Sequelize.UUIDV4,
+    },
+    firstName: {
+      type: Sequelize.UUID,
```
Comment (acd37) [inline src/models/User.js:9]: `.STRING?`

File: src/models/User.js [hunk 3]
```diff
@@ -1,5 +1,29 @@
...
+      primaryKey: true,
+      defaultValue: Sequelize.UUIDV4,
+    },
+    lastName: {
+      type: Sequelize.UUID,
```
Comment (acd37) [inline src/models/User.js:14]: `.STRING`?

**Comment by acd37** [comment]
@petewanca left a couple of comments to review

File: client/src/App.js (hunk 1 above)
Comment (petewanca) [reply client/src/App.js:3]: got it

File: src/models/User.js (hunk 2 above)
Comment (petewanca) [reply src/models/User.js:9]: WHOOPS

File: src/models/User.js (hunk 3 above)
Comment (petewanca) [reply src/models/User.js:14]: whoops again

**Comment by petewanca** [comment]
fixing and will open new PR

**Pull request closed without merging**